dev/io.openliberty.microprofile.telemetry.internal_fat.common/src/io/openliberty/microprofile/telemetry/internal_fat/shared/connectivity/ConnectionErrorWatcher.java

```diff
@@ -0,0 +1,244 @@
+/*******************************************************************************
+ * Copyright (c) 2024 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+package io.openliberty.microprofile.telemetry.internal_fat.shared.connectivity;
+
+import static org.junit.Assert.fail;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import org.junit.internal.AssumptionViolatedException;
+import org.junit.rules.TestRule;
+import org.junit.runner.Description;
+import org.junit.runners.model.Statement;
+
+import com.ibm.websphere.simplicity.log.Log;
+
+import componenttest.topology.impl.LibertyServer;
+
+/**
+ * Ignore test failures which occur due to connectivity errors between liberty and the system collecting spans.
+ * <p>
+ * Test failures of monitored tests will be ignored if there are errors in the server log similar to
+ *
+ * <pre>
+ * WARN io.opentelemetry.exporter.internal.grpc.GrpcExporter - Failed to export spans.
+ * </pre>
+ *
+ * <p>
+ * However, if more than a certain percentage of monitored tests fail in this way, an error will be reported at the end of the suite.
+ * <p>
+ * In the suite class:
+ *
+ * <pre>
+ * &#64;ClassRule
+ * public static TestRule connectionErrorWatcher = new ConnectionErrorWatcher(0.05);
+ * </pre>
+ *
+ * <p>
+ * In the test class:
+ *
+ * <pre>
+ * &#64;Server("MyServer")
+ * public static LibertyServer server;
+ *
+ * &#64;Rule
+ * public TestRule ignoreConnectionErrors = FATSuite.connectionErrorWatcher.ignoreConnectivityFailures(server);
+ * </pre>
+ */
+public class ConnectionErrorWatcher implements TestRule {
+
+    private final List<Failure> failures = new ArrayList<>();
+    private double failureRatioThreshold;
+    private int testsRun = 0;
+    private boolean watcherActive = false;
+
+    /**
+     * Create a new connection error watcher.
+     * <p>
+     * The watcher is created with the default failure ratio threshold of 5%
+     */
+    public ConnectionErrorWatcher() {
+        super();
+        failureRatioThreshold = 0.05;
```

Comment:
I would suggest changing this to 

`failureRatioThreshold = (System.getProperty("personalBuild") == null) ? 0.05 : 0.00`

I found that property in SOEHttpPostUtil.java. 

That way if we create a legitimate product bug that causes `GrpcExporter - Failed to export spans` we'll see it even if only one test catches it. 

(I plan to verify this by running a test PR that asserts that property is null and see if I get lots of failures)

Reply:
https://github.com/OpenLiberty/open-liberty/pull/29462 - `System.getProperty("personalBuild")` works as advertised.